posts/apache-spark-3.0.md

```diff
@@ -0,0 +1,54 @@
+- 2x performance improvement on TPC-DS over Spark 2.4
+- Simpler/better PySpark exceptions
+- Up to 40x speedups for calling R user-defined functions
+- Structured Streaming UI
+
+Check the [migration guide](https://spark.apache.org/docs/3.0.0/migration-guide.html) in order to switch your code to the new version and also note that no major code changes should be required.
+
+# Python API improvements
+
+I've been always a fan of Python for its simplicity, libraries and capacity of doing more with less code. [PySpark](https://spark.apache.org/docs/latest/api/python/index.html) is great, especially if you want to use ML or NLP libraries, but the overall performances are always gonna be better on a Scala environment.
+
+Recently, Python passed Scala on the most used language on Spark which is not a surprise since PySpark has more than 5 million monthly downloads on PyPI. The goal of this release was to improve the functionalities and usability of PySpark. That includes a redesign of the Pandas UDF API using Python type hints, new Pandas UDF types, and more Pythonic error handling. Pandas is still my most favorite library for data analysis/manipulation but it is a single-node processing library which doesn't really fit in a distributed environment. I recommend using [Koalas](https://koalas.readthedocs.io/en/latest/) which is basically the Pandas API on Apache Spark. The Koalas API gives a coverage close to 80% of the Pandas API which is enough for most cases.
+
+PySpark errors are not always user-friendly to Python users because Spark was originally designed to run on a JVM. This release will simplify PySpark exceptions by making them more Pythonic and will also avoid showing unnecessary JVM stack trace.
+
+# Spark SQL engine
+
+Most Spark applications are using the Spark SQL engine (based on Databricks stats) and one of the goal of this release was to improve performances and ANSI compatibility. Spark 3.0 will perform around 2x faster than a Spark 2.4 environment in the total runtime. In order to improve performances and query tuning a new framework was introduced: [Adaptive Query Execution (AQE)](https://databricks.com/blog/2020/05/29/adaptive-query-execution-speeding-up-spark-sql-at-runtime.html).
+
+AQE will generate a better execution plan at runtime which is crucial because of the separation between storage and compute in Spark (which make data unpredictable). In other words, AQE will create a better plan even if your data are absent or inaccurate (the initial Spark plan would have been inadequate).
+
+The ANSI compatibility aims to make workload migration from other SQL engines easier by improving the SQL compliance. Spark will forbid you the use of SQL reserved keywords (LIKE, IN, SELECT, etc) to avoid having issues later on. Also, when inserting data in a predefined schema, Spark will validate the data types to improve data quality. The property `spark.sql.ansi.enabled` must be set to `true` in order to enable this feature.
```

Comment:
```suggestion
The ANSI compatibility aims to make workload migration from other SQL engines easier by improving the SQL compliance. Spark will forbid you to use SQL reserved keywords (LIKE, IN, SELECT, etc) to avoid having issues later on. Also, when inserting data in a predefined schema, Spark will validate the data types to improve data quality. The property `spark.sql.ansi.enabled` must be set to `true` in order to enable this feature.
```